fixtures/templateloader.py

```diff
@@ -2,22 +2,32 @@
 """Preloads all templates on all providers that were selected for testing. Useful for test collect.
 """
 import pytest
-from fixtures.pytest_store import store
+
 from cfme.utils import trackerbot
+from cfme.utils.conf import env
 from cfme.utils.providers import list_provider_keys
+from fixtures.pytest_store import store
 
 TEMPLATES = {}
 
 
 @pytest.mark.tryfirst
 def pytest_addoption(parser):
     # Create the cfme option group for use in other plugins
-    parser.addoption("--use-template-cache", dest="use_template_cache", action="store_true",
-        default=False, help="Use a cached version of the templates and not redownload them")
+    parser.addoption(
+        "--use-template-cache", dest="use_template_cache", action="store_true",
+        default=False, help="Use a cached version of the templates and not redownload them"
+    )
 
 
 def pytest_configure(config):
-    if store.parallelizer_role == 'master' or trackerbot.conf.get('url') is None:
+    is_dev = False
+    if 'appliances' in env:
+        for appliance in env.appliances:
+            if appliance.get('is_dev', False):
```

Comment:
**QUESTION:**  It looks like you are seitting the appliance to be a dev appliance if it wasn't set that way.   Why do that?   Probably the answer is worthy of being codified in a comment.